src/tests/unit/status-badge.test.tsx

```diff
@@ -0,0 +1,170 @@
+import { configureStore, combineReducers } from "@reduxjs/toolkit";
+import React, { ReactNode } from "react";
+import { Provider } from "react-redux";
+import { afterEach, beforeEach, describe, expect, test, vi } from "vitest";
+import { cleanup, render } from "@testing-library/react";
+import configReducer from "../../core/config.slice";
+import StatusBadge from "../../apps/loan-list/materials/utils/status-badge";
+
+const store = configureStore({
+  reducer: combineReducers({
+    config: configReducer
+  }),
+  preloadedState: {
+    config: {
+      data: {
+        expirationWarningDaysBeforeConfig: "6"
+      }
+    }
+  }
+});
+
+const Wrapper = ({ children }: { children: ReactNode }) => (
+  <Provider store={store}> {children} </Provider>
+);
+describe("Status Badge tests", () => {
+  afterEach(() => {
+    cleanup();
+    vi.useRealTimers();
+  });
+
+  beforeEach(() => {
+    vi.useFakeTimers();
+    const date = new Date("2023-11-09T12:00:00+0100");
+    vi.setSystemTime(date);
+  });
+
+  test("Should not render anything if no props are given", () => {
+    const { getByTestId } = render(
+      <Wrapper>
+        <div data-testid="badge-wrapper">
+          <StatusBadge />
+        </div>
+      </Wrapper>
+    );
+
+    const html = getByTestId("badge-wrapper");
+
+    expect(html).toMatchInlineSnapshot(`
+    <div
+      data-testid="badge-wrapper"
+    />
+  `);
+  });
+
+  test("Expiring soon warning if within threshold", () => {
+    const { getByTestId } = render(
+      <Wrapper>
+        <div data-testid="badge-wrapper">
+          <StatusBadge
+            showBadgeWithDueDate
+            // Since the due date is tomorrow and the warning threshold is set to within 6 days
+            // we should get a warning.
+            badgeDate="2023-11-10 12:00:00+0100"
```

Comment:
:+1: I appreciate this comment.

One might argue that you could make relationship in code calculate the date as `date` - 1 day but all these programmatically set values can also make it harder to read the test case. Having the comment helps me understand the relationship in text.

Reply:
👍